File: surveyaugur_webapp/webapp/src/app/home/questiongroups/services/question.ts
import {UUID} from "angular2-uuid";
import {QuestionType} from "./questionType";

export class Question{
  id: UUID
  questionBody: string;
  tags: string[];
  questionType: QuestionType; // MULTIPLE SELECT, SINGLE SELECT, SUBJECTIVE
  options: string[];
  questionGroup_id: UUID; //NOT NULL
  createdBy: UUID;
  createdOnTS: Date;
  updatedOnTS: Date;
  updatedBy:String;
}

File: survey_authoring/src/main/java/com/stackroute/survey_authoring/Models/Question.java
package com.stackroute.survey_authoring.Models;



import com.stackroute.survey_authoring.Enums.QuestionType;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Date;
import java.util.UUID;

@Data
@NoArgsConstructor
public class Question implements Serializable {

    private UUID id;
    private String questionBody;
    private String[] tags;
    private QuestionType questionType;
    private String []options;
    private UUID questionGroupId;
    private UUID createdBy;
    private UUID updatedBy;
    private Date createdOn;
    private Date updatedOn;
}
